src/Pages/Main/Main.js

```diff
@@ -0,0 +1,78 @@
+import { useState, useEffect } from "react";
+import SliderComponent from "../../Components/SliderComponent";
+import MainSlider from "./components/MainSlider";
+import InfiniteScroll from "../../Components/InfiniteScroll";
+import styled from "styled-components";
+import SubBanner from "./components/SubBanner";
+
+const Main = ({ theme }) => {
+  const [top10, setTop10] = useState([]);
+  const [planned, setPlanned] = useState([]);
+  const [updated, setUpdated] = useState([]);
+
+  useEffect(() => {
+    fetch("http://localhost:3000/Data/data.json")
+      .then((res) => res.json())
+      .then((res) => {
+        setTop10(res.top_10_data);
+        setPlanned(res.planned_data);
+        setUpdated(res.updated_data);
+      });
+  }, []);
+
+  const handleScroll = () => {
+    const scrollHeight = document.documentElement.scrollHeight;
+    const scrollTop = document.documentElement.scrollTop;
+    const clientHeight = document.documentElement.clientHeight;
+    if (scrollTop + clientHeight >= scrollHeight - 50) {
+      if (updated.length > 50) {
+        return;
+      }
+      setUpdated([...updated, ...updated]);
+    }
+  };
```

Comment:
```suggestion
    if (scrollTop + clientHeight >= scrollHeight - 50) {
      if (updated.length > 50) return
      
      setUpdated([...updated, ...updated]);
    }
  };
```
- if문 한줄로 표현 가능할 경우 이렇게 써주는 것도 좋아보입니다
- updated를 두번 넣는건가요?